gtk4/src/expression.rs

```diff
@@ -182,6 +182,108 @@ impl glib::value::ToValueOptional for Expression {
     }
 }
 
+// rustdoc-stripper-ignore-next
+/// Trait containing convenience methods in creating [`gtk::PropertyExpression`] that
+/// looks up a property of a [`glib::Object`].
+///
+/// # Example
+///
+/// `label_expression` is a [`gtk::Expression`] that looks up at Button's `label`
+/// property.
+///
+/// ```
+/// let button = Button::new();
+/// button.set_label("Label property");
+///
+/// let label_expression = button.property_expression("label");
+/// ```
+pub trait GObjectPropertyExpressionExt {
+    // rustdoc-stripper-ignore-next
+    /// Create an expression looking up an object's property
+    fn property_expression(&self, property_name: &str) -> crate::PropertyExpression;
+
+    // rustdoc-stripper-ignore-next
+    /// Create an expression looking up an object's property with a weak reference.
+    fn weak_property_expression(&self, property_name: &str) -> crate::PropertyExpression;
+}
+
+impl<T: IsA<glib::Object>> GObjectPropertyExpressionExt for T {
+    fn property_expression(&self, property_name: &str) -> crate::PropertyExpression {
+        let obj_expr = crate::ConstantExpression::new(self);
+        crate::PropertyExpression::new(T::static_type(), Some(&obj_expr), property_name)
+    }
+
+    fn weak_property_expression(&self, property_name: &str) -> crate::PropertyExpression {
+        let obj_expr = crate::ObjectExpression::new(self);
+        crate::PropertyExpression::new(T::static_type(), Some(&obj_expr), property_name)
+    }
+}
+
+// rustdoc-stripper-ignore-next
+/// Trait containing convenience methods to create chains of [`gtk::PropertyExpression`]
+/// or [`gtk::ClosureExpression`].
+///
+/// # Examples
+///
+/// Create an expression describing `list_item->item->number` and bind it to button's label.
+///
+/// ```
+/// let factory = SignalListItemFactory::new();
+/// factory.connect_setup(|_, list_item| {
+///     let button = Button::new(None);
+///     list_item.set_child(Some(&button));
+///
+///     list_item.property_expression("item")
+///         .property_expression("number")
+///         .bind(&button, "label", Some(&button));
+/// });
+/// ```
+///
+/// Same thing as above but adds a string prefix to the number.
+///
+/// ```
+/// let factory = SignalListItemFactory::new();
+/// factory.connect_setup(|_, list_item| {
+///     let button = Button::new(None);
+///     list_item.set_child(Some(&button));
+///
+///     list_item.property_expression("item")
+///         .property_expression("number")
+///         .closure_expression(|args| {
+///             let number: i32 = args[1].get().unwrap();
+///             format!("Number: {}", number)
+///         })
+///         .bind(&button, "label", Some(&button));
+/// });
+/// ```
+pub trait ExpressionChainExt {
```

Comment:
Now instead of a trait, can this just be in `impl Expression` and then it works the same?

Reply:
I agree with it, but changing the methods name to make it more clear

Reply:
Done :D